lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Sequential.hs

```diff
@@ -568,13 +574,76 @@ instance
     => NFData (SeqState n k)
 
 instance PersistPublicKey (k 'AccountK) => Buildable (SeqState n k) where
-    build (SeqState intP extP chgs _) = "SeqState:\n"
+    build (SeqState intP extP chgs _ path) = "SeqState:\n"
+        <> indentF 4 ("Derivation prefix: " <> build (toText path))
         <> indentF 4 (build intP)
         <> indentF 4 (build extP)
         <> indentF 4 ("Change indexes: " <> indentF 4 chgsF)
       where
         chgsF = blockListF' "-" build (pendingIxsToList chgs)
 
+newtype DerivationPrefix = DerivationPrefix
```

Comment:
Maybe it's obvious what this is to others, but some documentation could still help?

Reply:
Definitely.

Reply:
Added:

```hs
-- | Each 'SeqState' is like a bucket of addresses associated with an 'account'.
-- An 'account' corresponds to a subset of an HD tree as defined in BIP-0039.
--
-- cardano-wallet implements two similar HD schemes on top of BIP-0039 that are:
--
-- - BIP-0044 (for so-called Icarus wallets)
-- - CIP-1815 (for so-called Shelley and Jormungandr wallets)
--
-- Both scheme works by considering 5 levels of derivation from an initial root
-- key (see also 'Depth' from Cardano.Wallet.Primitive.AddressDerivation). A
-- SeqState keeps track of indexes from the two last levels of a derivation
-- branch. The 'DerivationPrefix' defines the first three indexes chosen for
-- this particular 'SeqState'.
```